src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -839,23 +842,22 @@ export default {
 		 * @param {boolean} [options.all] Whether all conversations should be fetched
 		 */
 		async handleShouldRefreshConversations(options) {
-			if (options?.all === true) {
-				if (this.isCurrentTabLeader) {
-					this.roomListModifiedBefore = 0
-					this.forceFullRoomListRefreshAfterXLoops = 10
-				} else {
-					// Force leader tab to do a full fetch
-					talkBroadcastChannel.postMessage({ message: 'force-fetch-all-conversations' })
-					return
-				}
-			} else if (options?.token && options?.properties) {
+			if (options?.token && options?.properties) {
 				await this.$store.dispatch('setConversationProperties', {
 					token: options.token,
 					properties: options.properties,
 				})
 			}
 
-			this.debounceFetchConversations()
+			if (this.isCurrentTabLeader) {
+				if (options?.all === true) {
+					this.roomListModifiedBefore = 0
+					this.forceFullRoomListRefreshAfterXLoops = 10
+				}
+				this.debounceFetchConversations()
+			} else {
+				talkBroadcastChannel.postMessage({ message: 'force-fetch-all-conversations', options })
```

Comment:
The signaling message is always received by one tab, right ? 
Because otherwise, there is a risk that tabs will resend the channel message multiple times and leader tab will receive all of them.

Reply:
Yes, as every tab counts as a different signaling session.
Only the tab that switches to conversation and receives that message from signaling, will post a broadcast message. Every other will receive it, but only leader tab will fetch the changes